python/rikai/spark/functions/vision.py

```diff
@@ -280,3 +281,22 @@ def spectrogram_image(
         format=image_format,
         **image_kwargs,
     )
+
+
+@udf(returnType=ArrayType(IntegerType()))
+def img_cluster(img_list, threshold=0.5, img_shape=(300, 300)):
+    from sklearn.cluster import AgglomerativeClustering
+    from skimage.metrics import structural_similarity as ssim
+
+    if len(img_list) == 1:
+        return [0]
+    clustering_model = AgglomerativeClustering(
+        n_clusters=None, distance_threshold=threshold, linkage="ward"
+    )
+    frames = [np.resize(img.to_numpy(), img_shape) for img in img_list]
```

Comment:
from documentation on numpy.resize: "This functionality is therefore not suitable to resize images, or data where each axis represents a separate and distinct entity."

Reply:
why call this "frames" as opposed to just images?

Reply:
will update using [skimage.transform.resize](https://scikit-image.org/docs/dev/api/skimage.transform.html#skimage.transform.resize)